From 759e65ea6d7db140e1506eb00cc932a4fdd1d7a8 Mon Sep 17 00:00:00 2001
From: Keerthy <j-keerthy@ti.com>
Date: Thu, 30 Apr 2020 12:10:17 +0530
Subject: [PATCH 13/28] arm: dts: k3-am65-main: Add pruss nodes for ICSSG2

Add pruss nodes. This is based 4.19 DT with interrupt properties
removed from pur/rtu nodes.

Signed-off-by: Keerthy <j-keerthy@ti.com>
[Jan: ported to mainline, removed psil properties]
Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 arch/arm/dts/k3-am65-main.dtsi | 91 ++++++++++++++++++++++++++++++++++
 1 file changed, 91 insertions(+)

diff --git a/arch/arm/dts/k3-am65-main.dtsi b/arch/arm/dts/k3-am65-main.dtsi
index b85e999e2e..4b5460ad24 100644
--- a/arch/arm/dts/k3-am65-main.dtsi
+++ b/arch/arm/dts/k3-am65-main.dtsi
@@ -368,4 +368,95 @@
 		#phy-cells = <0>;
 		ti,dis-chg-det-quirk;
 	};
+
+	icssg2: icssg@b200000 {
+		compatible = "ti,am654-icssg";
+		power-domains = <&k3_pds 64 TI_SCI_PD_EXCLUSIVE>;
+		reg = <0 0xb200000 0 0x80000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		dma-ranges;
+
+		icssg2_mem: memories@b200000 {
+			reg = <0xb200000 0x2000>,
+			      <0xb202000 0x2000>,
+			      <0xb210000 0x10000>;
+			reg-names = "dram0", "dram1",
+				    "shrdram2";
+		};
+
+		icssg2_cfg: cfg@b226000 {
+			compatible = "syscon";
+			reg = <0xb226000 0x200>;
+		};
+
+		icssg2_iep: iep@b22e000 {
+			compatible = "syscon";
+			reg = <0xb22e000 0x1000>;
+		};
+
+		icssg2_mii_rt: mii-rt@b232000 {
+			compatible = "syscon";
+			reg = <0xb232000 0x100>;
+		};
+
+		icssg2_mii_g_rt: mii-g-rt@b233000 {
+			compatible = "syscon";
+			reg = <0xb233000 0x1000>;
+		};
+
+		icssg2_intc: interrupt-controller@b220000 {
+			compatible = "ti,am654-icssg-intc";
+			reg = <0xb220000 0x2000>;
+			interrupt-controller;
+			#interrupt-cells = <1>;
+		};
+
+		pru2_0: pru@b234000 {
+			compatible = "ti,am654-pru";
+			reg = <0xb234000 0x4000>,
+			      <0xb222000 0x100>,
+			      <0xb222400 0x100>;
+			reg-names = "iram", "control", "debug";
+			firmware-name = "am65x-pru2_0-fw";
+		};
+
+		rtu2_0: rtu@b204000 {
+			compatible = "ti,am654-rtu";
+			reg = <0xb204000 0x2000>,
+			      <0xb223000 0x100>,
+			      <0xb223400 0x100>;
+			reg-names = "iram", "control", "debug";
+			firmware-name = "am65x-rtu2_0-fw";
+		};
+
+		pru2_1: pru@b238000 {
+			compatible = "ti,am654-pru";
+			reg = <0xb238000 0x4000>,
+			      <0xb224000 0x100>,
+			      <0xb224400 0x100>;
+			reg-names = "iram", "control", "debug";
+			firmware-name = "am65x-pru2_1-fw";
+		};
+
+		rtu2_1: rtu@b206000 {
+			compatible = "ti,am654-rtu";
+			reg = <0xb206000 0x2000>,
+			      <0xb223800 0x100>,
+			      <0xb223c00 0x100>;
+			reg-names = "iram", "control", "debug";
+			firmware-name = "am65x-rtu2_1-fw";
+		};
+
+		icssg2_mdio: mdio@b232400 {
+			compatible = "ti,davinci_mdio";
+			reg = <0xb232400 0x100>;
+			clocks = <&k3_clks 64 3>;
+			clock-names = "fck";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			bus_freq = <1000000>;
+			status = "disabled";
+		};
+	};
 };
-- 
2.25.1

